### PR TITLE
added props

### PR DESCRIPTION
**Pull request by JamesMDunn**

### CODE REVIEW

File: crates/rabbitmq/src/queue_type.rs [hunk 1]
```diff
@@ -115,13 +115,14 @@ impl<'a> QueueInfo<'a> {
...
         routing: Option<&str>,
+        props: Option<&BasicProperties>,
     ) -> Result<PublisherConfirm> {
         chan.basic_publish(
             self.0.exchange.as_ref(),
             routing.unwrap_or(""),
             BasicPublishOptions::default(),
             data,
-            BasicProperties::default(),
+            props.unwrap_or(&BasicProperties::default()).clone(),
```
Comment (JamesMDunn) [inline crates/rabbitmq/src/queue_type.rs]: Would there be a better approach than doing `clone`here?

File: crates/rabbitmq/src/producer.rs [hunk 2]
```diff
@@ -42,6 +42,21 @@ where
         &self,
         val: impl std::borrow::Borrow<Q::Message>,
         routing_key: Option<&str>,
+    ) -> Result<()> {
+        let _ = &self.write_with_props(val, routing_key, None).await;
```
Comment (thesoftwarejedi) [inline crates/rabbitmq/src/producer.rs]: ouch. you're swallowing any errors.

Comment (thesoftwarejedi) [reply crates/rabbitmq/src/producer.rs]: can just be 1 line, no semicolon
`&self.write_with_props(val, routing_key, None)`
and remove async from function

Comment (JamesMDunn) [reply crates/rabbitmq/src/producer.rs]: oof okay. nice catch!

**Review by thesoftwarejedi — APPROVED** [review]

**Pull request merged**